source/robotstxtparser.php

```diff
@@ -79,14 +79,18 @@ class RobotsTxtParser
 
     // robots.txt file content
     private $content = '';
-
+    /**
+     * @var array
+     */
+    private $regExpPatternReplace;
+    
     /**
      * Constructor
      *
      * @param string $content - file content
      * @param string $encoding - encoding
      */
-    public function __construct($content, $encoding = self::DEFAULT_ENCODING)
+    public function __construct($content, $encoding = self::DEFAULT_ENCODING, array $regExpPatternReplace = array())
```

Comment:
If you would like to pass in your own replaces, I guess it would make sense to merge our expected replaces with the user-ones, or at least pass the default ones by default:


```suggestion
    public function __construct($content, $encoding = self::DEFAULT_ENCODING, array $regExpPatternReplace = array('@' => '\@'))
```